Frontend/pages/profile.tsx

```diff
@@ -25,6 +28,32 @@ import {
 
 const Home = () => {
   const { currentAccount, connectWallet } = useContext(AuthContext);
+  const { myDonations, isLoadingUserDonations } = useFundraiserItems();
+  const [totalDonations, setTotalDonations] = useState('');
+
+  const getTotalDonations = async () => {
+
+    const items = await Promise.all(
+      myDonations.map(async (item) => {
+         return item.userDonations
+      }))
+
+       
+      const donations: any[] = [];
+
+      // @ts-ignore TODO: fix typescript error
+      items.map((item) => { return item?.map(res => donations.push(res.donationAmount)); })
+
+      setTotalDonations(donations.reduce((a, b) => Number(a) + Number(b), 0).toFixed(2));
+  }
+  
+
+  useEffect(() => {
+    if(myDonations){
+      getTotalDonations();
+    }
+  }, [myDonations])
```

Comment:
All that logic can be moved and taken from the dedicated hook or context, so once we isolate that part according to https://github.com/paschal533/FundKyiv/pull/5/files#r984349565, it should became much cleaner